### ISSUE TITLE
Probably invalid usage of `ParamSpec` in argument type of generic function

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#2534]
**To Reproduce**
```python3
#pyright: strict

from typing_extensions import ParamSpec, Protocol

P = ParamSpec("P")

class SomeWrapper(Protocol[P]):
    def __call__(self, *args: P.args, **kwargs: P.kwargs):
        ...

    # no errors, but this extra int is probably wrong
    def use_some_wrapper(self, x: "SomeWrapper[P, int]"):
        ...


 # also no errors with generic standalone function
def use_some_wrapper(x: SomeWrapper[P, int]):
    pass


class X(SomeWrapper[str]):
    def __call__(self, a: str):
        pass

# interestingly, that extra 'int' in SomeWrapper.use_some_wrapper is nowhere to be found
reveal_type(X().use_some_wrapper)
# info: Type of "X().use_some_wrapper" is "(x: SomeWrapper[(str)]) -> None"
```

**Expected behavior**
I think `ParamSpec`s and normal `TypeVar`s are not supposed to mix outside of `Concatenate` ?

**VS Code extension or command-line**
pyright extension version 1.1.184

**Comment by erictraut** [comment]
Yeah, this should probably be flagged as an error.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.185, which I just published. It will also be included in the next release of pylance.
